core/iwasm/libraries/lib-socket/src/wasi/wasi_socket_ext.c

```diff
@@ -53,6 +53,34 @@ sockaddr_to_wasi_addr(const struct sockaddr *sock_addr, socklen_t addrlen,
     return ret;
 }
 
+static __wasi_errno_t
+wasi_addr_to_sockaddr(const __wasi_addr_t *wasi_addr,
+                      struct sockaddr *sock_addr, socklen_t *addrlen)
+{
+    switch (wasi_addr->kind) {
+        case IPv4:
+        {
+            struct sockaddr_in sock_addr_in = { 0 };
+
+            sock_addr_in.sin_family = AF_INET;
+            sock_addr_in.sin_addr.s_addr = (wasi_addr->addr.ip4.addr.n3 << 24)
+                                           | (wasi_addr->addr.ip4.addr.n2 << 16)
+                                           | (wasi_addr->addr.ip4.addr.n1 << 8)
+                                           | wasi_addr->addr.ip4.addr.n0;
+            sock_addr_in.sin_port = htons(wasi_addr->addr.ip4.port);
+            memcpy(sock_addr, &sock_addr_in, sizeof(sock_addr_in));
+
+            *addrlen = sizeof(sock_addr_in);
+        } break;
```

Comment:
```suggestion
            break;
        }
```

Reply:
Hi @lum1n0us please see my previous comment above:

> Please note if I make that change the build fails: https://github.com/bytecodealliance/wasm-micro-runtime/runs/7536836808?check_suite_focus=true and the error is:
> 
> ```
> --- core/shared/platform/common/posix/posix_socket.c failed on code style checking.
> ```
> 
> As I said, clang-format-12 enforces this kind of styling; we can change the configuration of clang formatter, but I think that's outside of the scope of this change, so I'd prefer not doing that here.

Reply:
Sorry, just noticed that. We might have a glitch in the clang-format now.

Reply:
Btw, may I have a look at the code causes the error?
"--- core/shared/platform/common/posix/posix_socket.c failed on code style checking." ?

Reply:
This is the code you proposed as a change (putting `break` in a new line)

Reply:
I am hoping
```
        case IPv4:
        {
            struct sockaddr_in sock_addr_in = { 0 };

            sock_addr_in.sin_family = AF_INET;
            sock_addr_in.sin_addr.s_addr = (wasi_addr->addr.ip4.addr.n3 << 24)
                                           | (wasi_addr->addr.ip4.addr.n2 << 16)
                                           | (wasi_addr->addr.ip4.addr.n1 << 8)
                                           | wasi_addr->addr.ip4.addr.n0;
            sock_addr_in.sin_port = htons(wasi_addr->addr.ip4.port);
            memcpy(sock_addr, &sock_addr_in, sizeof(sock_addr_in));

            *addrlen = sizeof(sock_addr_in);
            break;
        }
```

Reply:
Ah, sorry I misinterpreted your initial comment. I'll move break inside the {} then.